### PR TITLE
feat(core): add role-application related APIs

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. separate role-user-related APIs from roles routes
2. add role-application related APIs, including:
    - GET /roles/:roleId/applications
    - POST /roles/:roleId/applications
    - DELETE /roles/:roleId/applications/:appId

## Testing

Secured by UTs and ITs.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] docs

OR

- [ ] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/core/src/queries/application.ts [hunk 1]
```diff
@@ -47,6 +54,43 @@ export const createApplicationQueries = (pool: CommonQueryMethods) => {
     return { count: Number(count) };
   };
 
+  const countM2mApplicationsByIds = async (applicationIds: string[]) => {
+    const { count } = await pool.one<{ count: string }>(sql`
+      select count(*)
+      from ${table}
+      where ${fields.type} = ${ApplicationType.MachineToMachine}
+      ${conditionalSql(
+        applicationIds,
```
Comment (wangsijie) [inline packages/core/src/queries/application.ts:63]: should throw error if ids is empty?

Comment (darcyYe) [reply packages/core/src/queries/application.ts]: Good call, I think can directly return the result if the `ids` is empty.

File: packages/core/src/routes/role.application.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            countM2mApplicationsByIds(applicationIds),
+            findM2mApplicationsByIds(limit, offset, applicationIds),
+          ]);
+
+          ctx.pagination.totalCount = count;
+          ctx.body = applications;
+
+          return next();
+        },
+        (error) => {
```
Comment (wangsijie) [inline packages/core/src/routes/role.application.ts:57]: This "tryThat" looks like for parsing search fields only

Comment (wangsijie) [reply packages/core/src/routes/role.application.ts:57]: `parseSearchParamsForSearch`

**Review by wangsijie — APPROVED** [review]
Overall LGTM

File: packages/core/src/queries/application.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/queries/application.ts:79]: But it seems that this code is deleted:
```
if (applicationIds.length === 0) {
  return { count: 0 };
}
```

Comment (darcyYe) [reply packages/core/src/queries/application.ts]: Forgot to push the latest changes, it should be good now.

File: packages/core/src/routes/role.application.ts (hunk 2 above)
Comment (darcyYe) [reply packages/core/src/routes/role.application.ts:60]: Fixed

**Pull request merged**